packages/office-ui-fabric-react/src/components/Rating/Rating.tsx

```diff
@@ -44,106 +70,72 @@ export class Rating extends BaseComponent<IRatingProps, IRatingState> {
   }
 
   public render() {
-    let stars: JSX.Element[] = [];
-    for (let i = this.props.min as number; i <= (this.props.max as number); ++i) {
-      stars.push(this._renderStar(i));
+    let stars = [];
+
+    for (let i = this.props.min as number; i <= (this.props.max as number); i++) {
+      let ratingStarProps: IRatingStarProps = {
+        fillPercentage: this._getFillingPercentage(i),
+        disabled: this.props.disabled ? true : false
+      };
+
+      stars.push(
+        <button
+          className={ css('ms-Rating-button', styles.ratingButton, {
+            ['ms-Rating--large ' + styles.rootIsLarge]: this.props.size === RatingSize.Large,
+            ['ms-Rating--small ' + styles.rootIsSmall]: this.props.size !== RatingSize.Large
+          }) }
+          key={ i }
+          { ...((i === Math.ceil(this.state.rating as number)) ? { 'data-is-current': true } : {}) }
+          onFocus={ this._onFocus.bind(this, i) }
+          disabled={ this.props.disabled || this.props.readOnly ? true : false }
+          role='presentation'
+        >
+          { this._getLabel(i) }
+          <RatingStar key={ i + 'rating' }  {...ratingStarProps} />
+        </button>
+      );
     }
 
     return (
       <div
-        className={ css('ms-Rating', this.props.className, {
-          ['ms-Rating--large ' + styles.rootIsLarge]: this.props.size === RatingSize.Large
-        }) }
-        role='application'
+        className={ css('ms-Rating-star') }
+        aria-label={ this.props.getAriaLabel ? this.props.getAriaLabel(this.state.rating ? this.state.rating : 0, this.props.max as number) : '' }
       >
-        <div className={ css('ms-Rating-container', styles.container) } role='radiogroup' aria-labelledby={ this.props.ariaLabelId }>
+        <FocusZone
+          direction={ FocusZoneDirection.horizontal }
+          tabIndex={ this.props.readOnly ? 0 : -1 }
```

Comment:
This is not a good idea. Why make the entire div focusable? it has no role or arialabel...

Reply:
This is what I originally suggested and even wrote a codepen to illustrate:

https://codepen.io/dzearing/pen/LzEKoN?editors=1111

Reply:
1) We want to focus on entire div in readonly mode for the narrator to read the rating value.
2) TabIndex -1 is used to avoid redundant code.